front/lib/swr/data_source_views.ts

```diff
@@ -174,25 +174,27 @@ export function useDataSourceViewDocument({
   dataSourceView,
   documentId,
   owner,
+  disabled,
 }: {
   dataSourceView: DataSourceViewType | null;
   documentId: string | null;
   owner: LightWorkspaceType;
+  disabled?: boolean;
 }) {
   const dataSourceViewDocumentFetcher: Fetcher<GetDataSourceViewDocumentResponseBody> =
     fetcher;
-  const disabled = !dataSourceView || !documentId;
+  const isDisabled = !dataSourceView || !documentId || disabled;
 
   const { data, error, mutate } = useSWRWithDefaults(
-    disabled
+    isDisabled
       ? null
       : `/api/w/${owner.sId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/documents/${encodeURIComponent(documentId)}`,
     dataSourceViewDocumentFetcher
   );
```

Comment:
Now that you have a real disabled, I think it's more solid to distinguish between being able to compute an url and disabling actively like this:

```ts
const url = dataSourceView && documentId ? `/api/w/${owner.sId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/documents/${encodeURIComponent(documentId)}` : null;

const { data, error, mutate } = useSWRWithDefaults(
 url,
 dataSourceViewDocumentFetcher,
 { disabled }
);
```